Ambit/Source/Ambit/Mode/ConfigImportExport.spec.cpp

```diff
@@ -1017,4 +1017,101 @@ void ConfigImportExportSpec::Define()
             JsonContent = nullptr;
         });
     });
+
+    Describe("OnExportMap()", [this]()
+    {
+        BeforeEach([this]()
+        {
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Set target platform for export
+            AmbitMode->UISettings->ExportPlatforms.SetWindows(true);
+
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+        });
+
+        It("Should Return FReply::Handled", [this]()
+        {
+            const FReply Reply = Exporter->OnExportMap();
+
+            TestTrue("Reply should be true on return", Reply.IsEventHandled());
+        });
+
+        AfterEach([this]()
+        {
+            GLevelEditorModeTools().DeactivateMode(FAmbitMode::EM_AmbitModeId);
+            Exporter = nullptr;
+            JsonContent = "";
+        });
+    });
+
+    Describe("OnExportGltf()", [this]()
+    {
+        BeforeEach([this]()
```

Comment:
This BeforeEach() implementation is identical to the one above. This repetition is a code smell indicating that these tests have something in common and should probably be grouped together in some way so that they can share a common BeforeEach(). Note that you can nest Describe() methods inside each other, each with its own BeforeEach() method.

Reply:
These are split up to differentiate between the two export options. Similar to the `OnExportSdf()` tests at the start. I can group them if we want, I have no preference.

Reply:
I can see the argument for the organization you chose, and I agree with it. I recently added a comment elsewhere about encapsulating this repetitive code into a function. If you do that it will allay my concerns.

Reply:
I have provided my reasoning on why helper functions haven't been introduced in another comment.